R/population-simulation-settings.R

```diff
@@ -0,0 +1,44 @@
+#' @title PopulationSimulationSettings
+#' @description  R6 class for Population Simulation Settings
+#' @field numberOfCores number of cores for parallel computation
+#' @field showProgress
+PopulationSimulationSettings <- R6::R6Class(
```

Comment:
Is there a reason why you are not using it the core object directly ? SimulationRunOptions

Reply:
The core object takes numberOfCoresToUse whereas the reporting engine one takes numberOfCores, which will be potentially used by mpi

Reply:
Hum... isn't it just a naming thing? The object are the same otherwise.

Reply:
Yes I can change RE to interpret numberOfCoresToUse as the numberOfCores for mpi.  The idea was to make sure _core_ can't read this new object (if input by mistake) which could cause it to set numberOfCoresToUse >1 in a parallel run, where we want numberOfCoresToUse = 1 and numberOfCores (for mpi) to be >1.  Can remove it if you think this is overkill.

Reply:
I think each node when using mpi will have one core anyways so the .NET runner will use the max number of core in the node at any rate which would be one. 
Let's leave it for now but if the role of this object is nothing more than being an exact copy of the CORE object, we should think about using it